python/pyiceberg/catalog/hive.py

```diff
@@ -0,0 +1,262 @@
+        except NoSuchObjectException as e:
+            # When the namespace doesn't exists, it throws the same error
+            raise NoSuchTableError(f"Table {table_name} cannot be found") from e
+
+    def drop_table(self, identifier: Union[str, Identifier]) -> None:
+        self._drop_table(identifier)
+
+    def purge_table(self, identifier: Union[str, Identifier]) -> None:
+        self._drop_table(identifier, True)
+
+    def rename_table(self, from_identifier: Union[str, Identifier], to_identifier: Union[str, Identifier]) -> Table:
+        from_database_name, from_table_name = self.identifier_to_tuple(from_identifier)
+        to_database_name, to_table_name = self.identifier_to_tuple(to_identifier)
+        try:
+            with self._client as open_client:
+                tbl = open_client.get_table(dbname=from_database_name, tbl_name=from_table_name)
+                tbl.dbName = to_database_name
+                tbl.tableName = to_table_name
+                open_client.alter_table(dbname=from_database_name, tbl_name=from_table_name, new_tbl=tbl)
+        except NoSuchObjectException as e:
+            raise NoSuchTableError(f"Table {from_database_name}.{from_table_name} not found") from e
+        except InvalidOperationException as e:
+            raise NoSuchNamespaceError(f"Destination database {to_database_name} does not exists") from e
+        return Table()
+
+    def create_namespace(self, namespace: Union[str, Identifier], properties: Optional[Properties] = None) -> None:
+        database_name = self.identifier_to_tuple(namespace)[0]
+
+        hive_database = HiveDatabase(name=database_name, parameters=properties)
+
+        try:
+            with self._client as open_client:
+                open_client.create_database(hive_database)
+        except AlreadyExistsException as e:
+            raise AlreadyExistsError(f"Database {database_name} already exists") from e
+
+    def drop_namespace(self, namespace: Union[str, Identifier]) -> None:
+        database_name = self.identifier_to_tuple(namespace)[0]
+        try:
+            with self._client as open_client:
+                open_client.drop_database(database_name, deleteData=True, cascade=False)
+        except InvalidOperationException as e:
+            raise NamespaceNotEmptyError(f"Database {database_name} is not empty") from e
+        except MetaException as e:
+            raise NoSuchNamespaceError(f"Database {database_name} does not exists") from e
+
+    def list_tables(self, namespace: Optional[Union[str, Identifier]] = None) -> List[Identifier]:
+        database_name = self.identifier_to_tuple(namespace)[0] if namespace else namespace
+        with self._client as open_client:
+            return open_client.get_all_tables(db_name=database_name)
+
+    def list_namespaces(self) -> List[Identifier]:
+        with self._client as open_client:
+            return open_client.get_all_databases()
```

Comment:
I think this is going to return a list of strings, not a list of identifiers.

Reply:
Good one, thanks!